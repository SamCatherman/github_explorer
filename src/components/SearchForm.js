import React from "react";
//destructured handlers passed to component from SearchContainer.
//Updates SearchContainer's state on input change and form submit.
const SearchForm = ({ handleFormSubmit, value, handleInputChange }) => (
  <form onSubmit={handleFormSubmit}>
    <input
      type="text"
      name="search"
      placeholder="find a github user..."
      value={value}
      onChange={handleInputChange}
    />
    <button className="button" type="submit">
      Find User
    </button>
  </form>
);

export default SearchForm;
