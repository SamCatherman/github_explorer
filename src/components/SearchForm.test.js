import React from "react";
import SearchForm from "./SearchForm";
import { shallow } from "enzyme";

describe("Search Form", () => {
  let wrapper = shallow(<SearchForm />);
  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
