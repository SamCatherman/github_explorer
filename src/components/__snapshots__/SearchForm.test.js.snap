// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="search"
          onChange={undefined}
          placeholder="find a github user..."
          type="text"
          value={undefined}
/>,
        <button
          className="button"
          type="submit"
>
          Find User
</button>,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "search",
          "onChange": undefined,
          "placeholder": "find a github user...",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find User",
          "className": "button",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Find User",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="search"
            onChange={undefined}
            placeholder="find a github user..."
            type="text"
            value={undefined}
/>,
          <button
            className="button"
            type="submit"
>
            Find User
</button>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "search",
            "onChange": undefined,
            "placeholder": "find a github user...",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find User",
            "className": "button",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Find User",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
